sdk/appconfiguration/app-configuration/test/internal/throttlingRetryPolicyTests.spec.ts

```diff
@@ -188,3 +192,57 @@ describe("ThrottlingRetryPolicy", () => {
     });
   });
 });
+
+describe("Should not retry forever - honors the abort signal passed", () => {
+  let client: AppConfigurationClient;
+  const connectionString = "Endpoint=https://myappconfig.azconfig.io;Id=key:ai/u/fake;Secret=abcd=";
+
+  beforeEach(function() {
+    if (!nock.isActive()) {
+      nock.activate();
+    }
+    nock("https://myappconfig.azconfig.io:443")
+      .persist()
+      .put(/.*/g)
+      .reply(
+        429,
+        {
+          type: "https://azconfig.io/errors/too-many-requests",
+          title: "Resource utilization has surpassed the assigned quota",
+          policy: "Total Requests",
+          status: 429
+        },
+        ["retry-after-ms", "123456"]
+      );
+
+    client = new AppConfigurationClient(connectionString);
+  });
+
+  afterEach(async function() {
+    nock.restore();
+    nock.cleanAll();
+    nock.enableNetConnect();
+  });
+
+  it("simulate the service throttling", async () => {
+    const key = generateUuid();
+    const numberOfSettings = 200;
+    const promises = [];
+    try {
+      for (let index = 0; index < numberOfSettings; index++) {
+        promises.push(
+          client.addConfigurationSetting(
+            {
+              key: key + "-" + index,
+              value: "added"
+            },
+            {
+              abortSignal: AbortController.timeout(1000)
+            }
+          )
+        );
+      }
+      await Promise.all(promises);
+    } catch (_) {}
```

Comment:
Eating the exception here doesn't seem right. How does this test things properly?

I think the only thing that would be valid here is potentially getting an AbortError, correct?

Reply:
Hmmm, will do.
Updated to check for abort error. Fine?

Reply:
I see. FINE.